<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<html>
<head>
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<title></title>
	<meta name="generator" content="LibreOffice 4.2.7.2 (Linux)">
	<meta name="created" content="20150223;155500546053521">
	<meta name="changed" content="20150312;145515054291531">
	<style type="text/css">
	<!--
		@page { margin: 0.79in }
		p { margin-bottom: 0.1in; line-height: 120% }
		h2.cjk { font-family: "Droid Sans Fallback" }
		h2.ctl { font-family: "FreeSans" }
		td p { margin-bottom: 0in }
		a:link { so-language: zxx }
	-->
	</style>
</head>
<body lang="en-US" dir="ltr">
<a id="linkers"></a>
<BR>
<p align="center" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 16pt"><u><b>Annotation pipeline documentation</b></u></font></font></p>


<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><b>1)Introduction</b></u></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">The
annotation pipeline is a tool that allows to structurally annotate
the protein coding parts of a genome through a combination of <i>ab
initio</i> gene predictions and homology searches. It is designed to
be run in the CNAG-cluster and it uses the JIP environment.</font></font></p>
<a id="chapter1"></a>


<a id="chapter2"></a>
<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><b>2)
Download and install </b></span></span><span style="font-style: normal"><span style="text-decoration: none"><b>the
following programs required</b></span></span><span style="font-style: normal"><span style="text-decoration: none"><b>
</b></span></span><span style="font-style: normal"><span style="text-decoration: none"><b>for</b></span></span><span style="font-style: normal"><span style="text-decoration: none"><b>
the pipeline:</b></span></span></font></font></p>
<ul>
Download and install JIP</span></span></b></font></font></h2>



<a href="http://pyjip.readthedocs.org/en/latest/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">http://pyjip.readthedocs.org/en/latest/</span></span></span></font></font></a></p> 


	<li><p align="justify" style="margin-bottom: 0in; line-height: 100%">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><span style="font-style: normal"><span style="font-weight: normal">Geneid:</span></span></u><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">
	</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Program
	used for </span></span></span><i><span style="text-decoration: none"><span style="font-weight: normal">ab
	initio </span></span></i><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">prediction
	of eukariotic genomic sequences. </span></span></span></font></font>
	</p>
	<ul>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Download
		and get installation instructions from:
		</span></span></span></font></font><a href="http://genome.crg.es/software/geneid/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">http://genome.crg.es/software/geneid/</span></span></span></font></font></a></p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		</p>
	</ul>
	<li><p align="justify" style="margin-bottom: 0in; line-height: 100%">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><span style="font-style: normal"><span style="font-weight: normal">Genemark-</span></span><span style="font-style: normal"><span style="font-weight: normal">E</span></span><span style="font-style: normal"><span style="font-weight: normal">T</span></span><span style="font-style: normal"><span style="font-weight: normal">:</span></span></u><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">
	</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Program
	used for </span></span></span><i><span style="text-decoration: none"><span style="font-weight: normal">ab
	initio </span></span></i><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">prediction
	of eukariotic genomic sequences.</span></span></span></font></font></p>
	<ul>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Download
		and get installation instructions from:
		</span></span></span></font></font><a href="http://exon.gatech.edu/GeneMark/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">http://exon.gatech.edu/GeneMark/</span></span></span></font></font></a></p>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">A</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">dd
		the GENEMARK_PATH to your environment before running the genemark
		or genemark-ET steps of the pipeline. (Example: export
		GENEMARK_PATH=/project/devel/aateam/src/GeneMark-ET/gmes_petap/ )</span></span></span></font></font></p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		</p>
	</ul>
	<li><p align="justify" style="margin-bottom: 0in; line-height: 100%">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><span style="font-style: normal"><span style="font-weight: normal">Augustus:</span></span></u><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">
	</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">P</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">rogram
	used for </span></span></span><i><span style="text-decoration: none"><span style="font-weight: normal">ab
	initio </span></span></i><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">prediction
	of eukariotic genomic sequences.</span></span></span></font></font></p>
	<ul>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Download
		and get installation instructions from:
		</span></span></span></font></font><u><a href="http://bioinf.uni-greifswald.de/augustus/downloads/index.php"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">http://bioinf.uni-greifswald.de/augustus/downloads/index.php</font></font></a></u></p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		</p>
	</ul>
	<li><p align="justify" style="margin-bottom: 0in; line-height: 100%">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><span style="font-style: normal"><span style="font-weight: normal">Glimmer</span></span><span style="font-style: normal"><span style="font-weight: normal">HMM</span></span><span style="font-style: normal"><span style="font-weight: normal">:</span></span></u><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">
	</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Program
	used for </span></span></span><i><span style="text-decoration: none"><span style="font-weight: normal">ab
	initio </span></span></i><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">prediction
	of eukariotic genomic sequences.</span></span></span></font></font></p>
	<ul>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Download
		and get installation instructions from:
		</span></span></span></font></font><u><a href="http://www.cs.jhu.edu/~genomics/GlimmerHMM/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">http://www.cs.jhu.edu/~genomics/GlimmerHMM/</font></font></a></u></p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		</p>
	</ul>
	<li><p align="justify" style="margin-bottom: 0in; line-height: 100%">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><span style="font-style: normal"><span style="font-weight: normal">PASA:</span></span></u><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">
	</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Program
	used to map RNA-seq data processed by cufflinks and ESTs from the
	same species or some close species to our genome of interest. </span></span></span></font></font>
	</p>
	<ul>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Download
		and get installation instructions from:
		</span></span></span></font></font><u><a href="http://pasa.sourceforge.net/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">http://pasa.sourceforge.net/</font></font></a></u></p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		</p>
	</ul>
	<li><p align="left" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">SPALN:
	</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Program
	used to map protein sequences from the same species or some close
	species to our genome of interest.</span></span></span></font></font></p>
	<ul>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Download
		and get installation instructions from:
		</span></span></span></font></font><u><a href="http://www.genome.ist.i.kyoto-u.ac.jp/~aln_user/spaln/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">http://www.genome.ist.i.kyoto-u.ac.jp/~aln_user/spaln/</font></font></a></u></p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		</p>
	</ul>
	<li><p align="left" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><span style="font-style: normal"><span style="font-weight: normal">EvidenceModeler</span></span><span style="font-style: normal"><span style="font-weight: normal">(EVM)</span></span><span style="font-style: normal"><span style="font-weight: normal">:</span></span></u><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">
	</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Program
	used to </span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">compute
	weighted consensus gene structure annotations based on the above
	</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">outputs.
	</span></span></span></font></font>
	</p>
	<ul>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">Download
		and get installation instructions from:
		</span></span></span></font></font><u><a href="http://evidencemodeler.sourceforge.net/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">http://evidencemodeler.sourceforge.net/</font></font></a></u></p>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">A</span></span></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">dd
		the EVM_PATH to your environment before running the pipeline .
		(Example:export export
		EVM_PATH=/project/devel/aateam/src/EVM_r2012-06-25/ )</span></span></span></font></font></p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		</p>
	</ul>
	<li><p align="justify" style="margin-bottom: 0in; line-height: 100%">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><span style="font-style: normal"><span style="font-weight: normal">BEDTOOLS:</span></span></u><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">
	Flexible tools for genome arithmetic and DNA sequence analysis. Used
	in some steps of the pipeline to compare the gene annotations made
	by different programs. </span></span></span></font></font>
	</p>
	<ul>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Download
		and get installation instructions from: </font></font>
		</p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		<font size="4" style="font-size: 14pt"><a href="http://bedtools.readthedocs.org/en/latest/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">http://bedtools.readthedocs.org/en/latest/</font></font></a></font></p>
		<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
		</p>
	</ul>
	<li><p align="justify" style="margin-bottom: 0in; line-height: 100%">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><u><span style="font-weight: normal">GMAP:</span></u></span><span style="font-style: normal"><span style="text-decoration: none"><span style="font-weight: normal">
	P</span></span></span>rogram for mapping and aligning cDNA sequences
	to a genome. Needed for running PASA.</font></font></p>
	<ul>
		<li><p align="justify" style="margin-bottom: 0in; line-height: 100%">
		<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Download
		and get installation instructions from:</font></font></p>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"><a href="http://research-pub.gene.com/gmap/"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">http://research-pub.gene.com/gmap/</font></font></a></p>
		<p align="justify" style="margin-bottom: 0in; line-height: 100%"></p>
	</ul>

<a href="#linkers"><CENTER><B>back</B></CENTER></a> <br>
<a id="chapter3"></a>

<p align="justify" style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal"><span style="text-decoration: none"><b>3)Detailed
explanation of each step </b></span></span>

<p align="justify" style="margin-bottom: 0in; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">The
CNAG annotation pipeline uses a<span lang="en-US"> combination of the
Program to Assemble Spliced Alignments </span><span lang="en-US">(PASA)</span><span lang="en-US">
and Evidence Modeler (EVM</span><span lang="en-US">) </span><span lang="en-US">to
obtain consensus coding sequence (CDS) models using three main
sources of evidence: aligned transcripts, aligned proteins, and gene
predictions.</span></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-weight: normal; line-height: 100%">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u><span style="font-style: normal">3.1</span><i>
Ab initio </i><span style="font-style: normal">gene predictions:</span></u></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Four
different <i>ab initio</i> gene predictors are included in the
pipeline. To run them it is needed to give as options the
species-specific parameter file that each tool needs and the path to
the genome masked for species specific repeats. The four gene
predictors contained in the pipeline are:</font></font></p>
<ul>
	<li><p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Geneid</font></font></p>
	<li><p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Augustus</font></font></p>
	<li><p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">GlimmmerHMM</font></font></p>
	<li><p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Genemark-ES</font></font></p>
</ul>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>3.2
Homology searches:</u></font></font></p>
<ul>
	<li><p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Program
	to Assemble Spliced Alignments (PASA) is used to map RNA-seq
	evidence models produced by cufflinks and transcript evidence in
	fasta format to the assembled genome. </font></font>
	</p>
	<li><p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">After
	running PASA it is possible to run the transdecoder step which finds
	coding regions in the transcripts. </font></font>
	</p>
	<li><p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
	<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span lang="en-US">Spaln
	is used to map protein evidence </span><span lang="en-US">in fasta
	format </span><span lang="en-US">to the genome. </span></font></font>
	</p>
</ul>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>3.3
Gene predictors with evidence:</u></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span lang="en-US">When
mapping the RNA-seq data to the genome assembly (step not included in
the pipeline yet), it is possible to get intron junctions to help the
</span><span lang="en-US"><i>ab initio</i></span><span lang="en-US">
gene predictors.</span></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span lang="en-US">Geneid,
Augustus </span><span lang="en-US">and Genemark-ET</span><span lang="en-US">
can be run </span><span lang="en-US">with these junctions as hints,
</span><span lang="en-US">which can help to produce better
</span><span lang="en-US">predictions.</span><span lang="en-US"> </span><span lang="en-US">Only
those junctions predicted in areas where we have coding evidence are
used. There are two ways of giving these junctions to the pipeline:
already processed for coding regions or directly all the junctions
outputted by Gem, in the latter case the pipeline runs a step of
junction processing,  in which it selects those junctions present in
spaln, geneid and augustus </span><span lang="en-US"><i>ab initio
</i></span><span lang="en-US">outputs. </span></font></font>
</p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>3.4
EVM:</u></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span lang="en-US">Once
all the previous steps have finished </span><span lang="en-US">the
pipeline combines </span><span lang="en-US">everything and produce</span><span lang="en-US">s</span><span lang="en-US">
consensus </span><span lang="en-US">coding sequence </span><span lang="en-US">models
</span><span lang="en-US">with</span><span lang="en-US">
EvidenceModeler </span><span lang="en-US">(EVM)</span><span lang="en-US">.
By default, the pipeline runs EVM with three different weight files
and after</span><span lang="en-US">wards</span><span lang="en-US"> it
selects the </span><span lang="en-US">output</span><span lang="en-US">
with better specificity and sensitivity </span><span lang="en-US">to</span><span lang="en-US">
the RNA-seq mappings. If desired </span><span lang="en-US">you can
</span><span lang="en-US">change the weights of each evidence step
and/or generate more or less weight files by modifying the json
configuration file. </span></font></font>
</p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>3.5
Annotation update:</u></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">The
consensus CDS models are then updated with UTRs and alternative exons
through two rounds of PASA’s annotation updates. A final round of
quality control is performed, fixing reading frames, intron phases
and removing some trancscripts with NMDs. The resulting transcripts
are then clustered into genes using shared splice sites or
significant sequence overlap as criteria for designation as the same
gene. Systematic identifiers with the prefix and version specified
with the “project-name” option of the json parameter file are
assigned to the genes, transcripts and protein products derived from
them.</font></font></p>
<a id="chapter4"></a>
<p align="justify" style="margin-bottom: 0in; font-style: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><b>4)
Creating the configuration file</b></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">The
python script <i>create_config_file.V01.py </i>needs to be run
previous to the annotation pipeline to generate the json
configuration file containing all the parameters that will be used
when running the pipeline. The parameters of the json configuration
file are divided into different categories.</font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">The
first compulsory argument is the “-jsonFile” option, that gives
the name of the configuration file to be written. </font></font>
</p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Next,
you can specify which jobs you do not want to run, if any, by saying
“- - no-stepname”. For example, if you specify “- -no-geneid”
it will run the whole pipeline except the geneid step. In this
General category it is also needed to give the pipeline Home
directory, which is where it will look for all the scripts. </font></font>
</p>
<p> </p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">General
			Parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-geneid
			          If specified, do not run geneid step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-augustus
			        If specified, do not run augustus step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-genemark
			        If specified, do not run genemark step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-glimmer
			         If specified, do not run glimmer step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-geneid-introns
			  If specified, do not run geneid with introns step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-augustus-introns
			</font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">If
			specified, do not run augustus with introns step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-genemark-ET
			     If specified, do not run genemark-ET step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-spaln
			           If specified, do not run spaln step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-pasa
			            If specified, do not run pasa step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-transdecoder
			    If specified, do not run transdecoder step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-evm
			             If specified, do not run EVM step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--no-update
			          If specified, do not run the annotation update step. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pipeline-HOME
			pipeline_HOME </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the pipeline home directory. Default </font></font>
			</p>
			<p align="justify">                       
			<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">/project/devel/aateam/src/Annotation_pipeline.V01/
			</font></font>
			</p>
		</td>
	</tr>
</table>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Once
you have decided which steps you want to run, there are some
compulsory parameters that need to be given for those steps. If you
try to run the script at this point it will show an error message
with all the the compulsory options that you need for those steps.
For example, if you run any <i>ab initio</i> gene predictor it will
ask you to give the masked genome and for the rest of steps it will
ask for the genome without masking.</font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Once
you have given all the input options, you can change some of the
default parameters. In the help of the <i>create_config_file.V01.py
</i>script you have a detailed information of all the parameters
classified by step and its default values. Type “
<i>create_configuration_file.</i><i>V01.</i><i>py -</i><i>h”.</i></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Important
to make sure that the input and output parameters point to the desire
location, check this in the json configuration file before running
the pipeline and change them by running again the
“create_config_file.V01.py” if needed. </font></font>
</p>
<p align="justify" style="margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>Defining
weights for the EVM step</u>: EVM<span lang="en-US"> combines </span><span lang="en-US">all
the gene predictions and homology searches and </span><span lang="en-US">produce</span><span lang="en-US">s</span><span lang="en-US">
consensus </span><span lang="en-US">coding sequence </span><span lang="en-US">models.
By default, the pipeline runs EVM with three different weight files
and after</span><span lang="en-US">wards</span><span lang="en-US"> it
selects the </span><span lang="en-US">output</span><span lang="en-US">
with better specificity and sensitivity </span><span lang="en-US">to</span><span lang="en-US">
the RNA-seq mappings. </span>It is possible to change the default
values of the weights by specifying the option “-
-stepname-weights”. For example, if you add the option “-
-augustus-weights 2 1 2”, the weights for augustus will be 2 in
weights_1.txt, 1 in weights_2.txt and 2 in weights_3.txt. </font></font>
</p>
<a id="chapter5"></a>
<p align="justify" style="margin-bottom: 0in; font-style: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><b>5)
Changing the spec file</b></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal">The
jip environment allows </span><span style="font-style: normal">the
definition of the</span><span style="font-style: normal">
</span><span style="font-style: normal">characteristics</span><span style="font-style: normal">
of each job that the pipeline </span><span style="font-style: normal">will
launch to the cluster</span><span style="font-style: normal">. It is
possible to change the default parameters by changing the
</span><i>annotation_pipeline.V01.spec </i><span style="font-style: normal">file.
</span><span style="font-style: normal">In t</span><span style="font-style: normal">here,
you can specify </span><span style="font-style: normal">values</span><span style="font-style: normal">
such as the queue where you want to launch each job, the number of
CPUs needed, the time, the priority, etc. </span><span style="font-style: normal">Dfeault
values for each job have been tested and seem to be the appropriate
ones to ensure the correct finalization of that job. </span></font></font>
</p>
<a id="chapter6"></a>
<p align="justify" style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><b><span style="font-style: normal">6</span><span style="font-style: normal">)</span><span style="font-style: normal">
Run</span><span style="font-style: normal">ning</span><span style="font-style: normal">
the </span><span style="font-style: normal">script
</span><span style="font-style: normal">annotation_pipeline.V01.jip</span></b></font></font></p>
<p align="justify" style="margin-bottom: 0in; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">If
the configuration and the spec file have been correctly defined,
everything should be ready to run the <i>annotation_pipeline.V01.jip
</i><span style="font-style: normal">script. </span><span style="font-style: normal">The
only option it takes is “-</span><span style="font-style: normal">c</span><span style="font-style: normal">”
which should contain the name of the configuration file. </span></font></font>
</p>
<p align="justify" style="margin-bottom: 0in; font-weight: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><span style="font-style: normal">N</span><span style="font-style: normal">ote
that some of the jobs will have as dependency some step of the
pipeline, so they will wait until their dependencies finish to start
running. </span></font></font>
</p>
<a href="#linkers"><CENTER><B>back</B></CENTER></a> <br>
<a id="chapter7"></a>
<p align="justify" style="margin-bottom: 0in; font-style: normal; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><b>
7)Examples</b></font></font></p>
<p></p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>Example
			for running the whole pipeline:</u></font></font></p>
			
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">./create_config_file.V01.py
			-jsonFile test.json --genome genome.scaffolds.fa --genome-masked
			genome.scaffolds.masked.fa –geneid-parameters
			training_geneid/H.sapiens.geneid.optimized.U12.param --proteins
			proteins.fasta –transcripts transcripts.fasta --pasa-config
			alignAssembly.config --pasadb pasa_db –cufflinks cufflinks.gtf
			–glimmer-directory
			training_glimmer/TrainGlimmM2014-11-13D18:10:20/ --species human
			--junctions junctions_all.gff –project-name HSAP 1A
			–extrinsic-file-augustus-introns extrinsic.M.RM.E.W.cfg
			–update-config annotCompare.config –junctions junctions.gff </font></font>
			</p>
			
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">./annotation_pipeline.V01.jip
			 -c test.json – submit</font></font></p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>
7.1 Running Augustus:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p></p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome-masked
			genome_masked </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta genome masked. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--species
			species     Species name to run augustus with its trained parameters.
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--augustus-prediction
			AUGUSTUS_PREDICTION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the augustus predictions. Default step 03_annotation_pipeline.V01//gene_predictions/augustus/
			augustus_gene_prediction.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--augustus-preEVM
			AUGUSTUS_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the augustus predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/augustus/augustus_preEVM.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Augustus
			parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--alternatives-from-sampling
			ALTERNATIVES_FROM_SAMPLING </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Report
			alternative transcripts generated through probabilistic,
			possible values true/false. Default true
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--uniqueGeneId
			UNIQUEGENEID </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">If
			true, output gene identifyers like this: seqname.gN,
			possible values true/false. Default true </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--gff3
			GFF3    </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">       Output in gff3 format, possible values ON/OFF.
			Default ON
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--sample
			SAMPLE  </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">     Default 60 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--noInFrameStop
			NOINFRAMESTOP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Do
			not report transcripts with in-frame stop codons. Otherwise,
			intron-spanning stop codons could occur, possible
			values true/false. Default true </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--maxtracks
			MAXTRACKS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Maximum
			number of tracks allowed. Default 2 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--singlestrand
			SINGLESTRAND </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Predict
			genes independently on each strand, allow overlapping
			genes on opposite strands. possible values true/false.
			Default false </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--strand
			STRAND   </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">    Possible values both/forward/backward. Default both </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--min-intron-len
			MIN_INTRON_LEN </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Default
			30 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--augustus-weights
			AUGUSTUS_WEIGHTS [AUGUSTUS_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to augustus predictions when running EVM.
			Specify the weight for each EVM run separated by a
			space. Example 2 2 1 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--additional-augustus-options
			ADDITIONAL_AUGUSTUS_OPTIONS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Additional
			augustus options to run it, see augustus help
			for more information about the possible options. </font></font>
			</p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.2
Running Geneid:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p></p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome-masked
			genome_masked </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta genome masked. </font></font>
			</p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-parameters
			geneid_parameters </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the geneid parameters file.</font></font></p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-prediction
			GENEID_PREDICTION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the geneid predictions. Default step03_annotation_pipeline.V01//gene_predictions/geneid/geneid.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-preEVM
			GENEID_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the geneid predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/geneid/geneid_preEVM.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Geneid
			parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-weights
			GENEID_WEIGHTS [GENEID_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to geneid predictions when running EVM. Specify
			the weight for each EVM run separated by a space.
			Example 2 1 2 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-options
			GENEID_OPTIONS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Desired
			geneid options to run it, see geneid documentation
			for more information. Default 3U </font></font>
			</p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.3
Running Glimmer:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome-masked
			genome_masked </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta genome masked. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--glimmer-directory
			glimmer_directory </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the directory containing the trained parameters
			for running glimmer. </font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--glimmer-prediction
			GLIMMER_PREDICTION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the glimmer predictions. Default step03_annotation_pipeline.V01//gene_predictions/glimmer.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--glimmer-preEVM
			GLIMMER_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the glimmer predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/glimmer_preEVM.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Glimmer
			parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--glimmer-weights
			GLIMMER_WEIGHTS [GLIMMER_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to glimmer predictions when running EVM.Specify
			the weight for each EVM run separated by a space.
			Example 1 1 1 </font></font>
			</p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.4
Running Genemark-ES:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
</p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome-masked
			genome_masked </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta genome masked. </font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">    <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genemark-prediction
			GENEMARK_PREDICTION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the genemark predictions. Default step03_annotation_pipeline.V01//gene_predictions/genemark.gtf
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genemark-preEVM
			GENEMARK_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the genemark predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/genemark_preEVM.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Genemark
			parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--min-contig
			MIN_CONTIG </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Will
			ignore contigs shorter then min_contig in training.
			Default 50000 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--max-contig
			MAX_CONTIG </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Will
			split input genomic sequence into contigs shorter than
			max_contig. Default 5000000 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--max-gap
			MAX_GAP   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">  Will split sequence at gaps longer than max_gap. Letters
			'n' and 'N' are interpreted as standing withingaps.
			Default 5000 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genemark-cores
			GENEMARK_CORES </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Number
			of threads for running genemark. Default 8 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--additional-genemark-options
			ADDITIONAL_GENEMARK_OPTIONS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Additional
			genemark options to run it, see genemark documentation
			for more information. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genemark-weights
			GENEMARK_WEIGHTS [GENEMARK_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to genemark predictions when runningEVM.
			Specify the weight for each EVM run separated by a
			space. Example 1 1 1 </font></font>
			</p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.5
Running Geneid with introns:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%">
</p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome-masked
			genome_masked </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta genome masked. </font></font>
			</p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-parameters
			geneid_parameters </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the geneid parameters file.</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--junctions
			junctions </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the junctions gff file to run gene predictors with
			introns. Do not needed if incoding junctions existent
			file is given .</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--incoding-junctions
			incoding_junctions </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the junctions in coding regions gff file to run
			gene predictors with introns. (Optional, it can begiven
			if this file already exists or if you want to keep
			the processed incoding junctions in that concrete path.) </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-preEVM
			GENEID_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the geneid predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/geneid/geneid_preEVM.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--augustus-preEVM
			AUGUSTUS_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the augustus predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/augustus/augustus_preEVM.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--spaln-cds
			SPALN_CDS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the spaln output in a cds gff3 format. Default
			step03_annotation_pipeline.V01//protein_and_transcript_mappings/spaln/proteins_spaln_cds.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-introns-prediction
			GENEID_INTRONS_PREDICTION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the geneid with introns predictions. </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Default
			step03_annotation_pipeline.V01//gene_predictions/geneid_with_introns/geneid_introns.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-introns-preEVM
			GENEID_INTRONS_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the geneid with introns predictions converted
			for EVM. Default step03_annotation_pipeline.V01//gene_predictions/geneid/geneid_preEVM.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Geneid
			Introns parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-introns-weights
			GENEID_INTRONS_WEIGHTS [GENEID_INTRONS_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to geneid with intron predictions when running
			EVM. Specify the weight for each EVM run separated
			by a space. Example 3 3 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-introns-options
			GENEID_INTRONS_OPTIONS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Desired
			geneid with intron options to run it, see geneid
			documentation for more information. Default 3U </font></font>
			</p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.6
Running Augustus with hints:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%"></p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome-masked
			genome_masked </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta genome masked. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--species
			species     Species name to run augustus with its trained parameters.
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--junctions
			junctions </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the junctions gff file to run gene predictors with
			introns. Do not needed if incoding junctions existent
			file is given .</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--incoding-junctions
			incoding_junctions </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the junctions in coding regions gff file to run
			gene predictors with introns. (Optional, it can be given
			if this file already exists or if you want to keep
			the processed incoding junctions in that concrete path.) </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-preEVM
			GENEID_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the geneid predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/geneid/geneid_preEVM.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--augustus-preEVM
			AUGUSTUS_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the augustus predictions converted forEVM.
			Default step03_annotation_pipeline.V01//gene_predictions/augustus/augustus_preEVM.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--spaln-cds
			SPALN_CDS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the spaln output in a cds gff3 format. </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Default
			step03_annotation_pipeline.V01//protein_and_transcript_mappings/spaln/proteins_spaln_cds.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--augustus-introns-prediction
			AUGUSTUS_INTRONS_PREDICTION </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Output
			file for the augustus with introns predictions. </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Default
			step03_annotation_pipeline.V01//gene_predictions/augustus_with_introns/augustus_introns.gff3
			</font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--augustus-introns-preEVM
			AUGUSTUS_INTRONS_PREEVM </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Output
			file for the augustus with introns predictions converted
			for EVM. Default step03_annotation_pipeline.V01//gene_predictions/augustus/augustus_preEVM.gff3
			</font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font size="3" style="font-size: 12pt">Augustus
			Introns parameters: </font>
			</p>
			<p align="justify" style="margin-bottom: 0in">
			<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--alternatives-from-sampling
			ALTERNATIVES_FROM_SAMPLING </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Report
			alternative transcripts generated through probabilistic,
			possible values true/false. Default true
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--uniqueGeneId
			UNIQUEGENEID </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">If
			true, output gene identifyers like this: seqname.gN,
			possible values true/false. Default true </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--gff3
			GFF3    </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">        Output in gff3 format, possible values ON/OFF.
			Default ON
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--sample
			SAMPLE    </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">    Default 60 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--noInFrameStop
			NOINFRAMESTOP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Do
			not report transcripts with in-frame stop codons.Otherwise,
			intron-spanning stop codons could occur, possible
			values true/false. Default true </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--maxtracks
			MAXTRACKS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Maximum
			number of tracks allowed. Default 2 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--singlestrand
			SINGLESTRAND </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Predict
			genes independently on each strand, allow overlapping
			genes on opposite strands. possible values true/false.
			Default false </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--strand
			STRAND   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">     Possible values both/forward/backward. Default both </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--min-intron-len
			MIN_INTRON_LEN </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Default
			30 </font></font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--augustus-introns-weights
			AUGUSTUS_INTRONS_WEIGHTS [AUGUSTUS_INTRONS_WEIGHTS ...] </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Weights
			given to augustus with intron predictions when running
			EVM. Specify the weight for each EVM run by a space. Example 3 3 3 </font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--additional-augustus-introns-options
			ADDITIONAL_AUGUSTUS_INTRONS_OPTIONS </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Desired
			augustus with intron options to run it, see augustus
			documentation for more information. </font>
			</p>
		</td>
	</tr>
</table>

<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.7
Running Genemark-ET:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%"></p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome-masked
			genome_masked </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta genome masked. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--junctions
			junctions </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the junctions gff file to run gene predictors 
with
			introns. Do not needed if incoding junctions 
existent
			file is given .</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--incoding-junctions
			incoding_junctions </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the junctions in coding regions gff file to 
run
			gene predictors with introns. (Optional, it can be 
given
			if this file already exists or if you want to keep
			the processed incoding junctions in that concrete path.) </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-preEVM
			GENEID_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the geneid predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/geneid/geneid_preEVM.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--augustus-preEVM
			AUGUSTUS_PREEVM </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the augustus predictions converted for EVM.
			Default step03_annotation_pipeline.V01//gene_predictions/augustus/augustus_preEVM.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--spaln-cds
			SPALN_CDS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the spaln output in a cds gff3 format. Default
			step03_annotation_pipeline.V01//protein_and_transcript_mappings/spaln/proteins_spaln_cds.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">    <font size="3" style="font-size: 12pt">--genemark-ET-prediction
			GENEMARK_ET_PREDICTION </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Output
			file for the genemark-ET predictions. Default step03_annotation_pipeline.V01//gene_predictions/genemark-ET.gtf
			</font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--genemark-ET-preEVM
			GENEMARK_ET_PREEVM </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Output
			file for the genemark-ET predictions converted for
			EVM. Default step03_annotation_pipeline.V01//gene_predictions/genemark-ET_preEVM.gff3 </font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font size="3" style="font-size: 12pt">Genemark-ET
			parameters: </font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--min-contig
			MIN_CONTIG </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Will
			ignore contigs shorter then min_contig in training.
			Default 50000 </font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--max-contig
			MAX_CONTIG </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Will
			split input genomic sequence into contigs shorter than
			max_contig. Default 5000000 </font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--max-gap
			MAX_GAP     Will split sequence at gaps longer than max_gap. Letters
			'n' and 'N' are interpreted as standing within gaps.
			Default 5000 </font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--genemark-cores
			GENEMARK_CORES </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Number
			of threads for running genemark. Default 8</font></p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--et-score
			ET_SCORE  </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">  Minimum score of intron in initiation of the ET algorithm.
			Default 4 </font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--additional-genemark-ET-options
			ADDITIONAL_GENEMARK_ET_OPTIONS </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Additional
			genemark-ET options to run it, see genemark documentation
			for more information. </font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--genemark-ET-weights
			GENEMARK_ET_WEIGHTS [GENEMARK_ET_WEIGHTS ...] </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Weights
			given to genemark-ET predictions when running EVM.
			Specify the weight for each EVM run separated by a
			space. Example 3 3 3 </font>
			</p>
		</td>
	</tr>
</table>

<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.8
Running Spaln:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">

</p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome
			genome     </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">   Path to the fasta genome.</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--proteins
			proteins   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"> Path to the fasta with protein evidence.</font></font></p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">    <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--spaln-gene
			SPALN_GENE </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the spaln output in a gene gff3 format.
			Default step03_annotation_pipeline.V01//protein_and_transcript_mappings/spaln/proteins_spaln_gene.gff3
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--spaln-cds
			SPALN_CDS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Output
			file for the spaln output in a cds gff3 format. Default
			step03_annotation_pipeline.V01//protein_and_transcript_mappings/spaln/proteins_spaln_cds.gff3
			</font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Spaln
			parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--ya
			YA       </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">         Stringency of splice site. 0-&gt;3: strong-&gt;weak.
			Default 1
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--M
			M       </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">           Number of outputs per query (max=4 for genome vs
			cDNA|protein).
			Default 4 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--O
			O        </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">          0:Gff3_gene; 1:alignment; 2:Gff3_match; 3:Bed;
			4:exon-inf;
			5:intron-inf; 6:cDNA; 7:translated; 8: block-only;
			12: binary. Default 0 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--Q
			Q      </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">            0:DP; 1-3: HSP-Search; 4-7; Block-Search.
			Default 7 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--t
			T    </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">              Number of threads. Default 8 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--spaln-weights
			SPALN_WEIGHTS [SPALN_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to spaln mappings when running EVM. Specify
			the weight for each EVM run separated by a space.
			Example 10 8 10 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--additional-spaln-options
			ADDITIONAL_SPALN_OPTIONS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Additional
			spaln options to run it, see spaln help for more
			information about the possible options. </font></font>
			</p>
		</td>
	</tr>
</table>

<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.9 Running PASA:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none"></p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome
			genome   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">     Path to the fasta genome.</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasadb
			pasadb   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">     Name of the pasa database, it must coincide with the name
			in pasa_config. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--transcripts
			transcripts </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta with transcript evidence. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasa-config
			pasa_config </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the pasa configuration file. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--cufflinks
			cufflinks </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the cufflinks gtf file. (Optional)</font></font></p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasa-dir
			PASA_DIR   Directory to keep all the pasa outputs. Default step03_annotation_pipeline.V01//protein_and_transcript_mappings/pasa/</font></font></p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Pasa
			parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasa-CPU
			PASA_CPU  </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">  Number of pasa_CPUs to run Pasa. Default 8 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasa-step
			PASA_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			from where to start running Pasa. Default 1 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasa-weights
			PASA_WEIGHTS [PASA_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to pasa mappings when running EVM. Specify
			the weight for each EVM run separated by a space.
			Example 8 10 8 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--create-database
			    If specified, create pasa database.</font></font></p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.10
Running transdecoder:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none"> </p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome
			genome   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">     Path to the fasta genome.</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasadb
			pasadb   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">     Name of the pasa database, it must coincide with the name
			in pasa_config. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--transcripts
			transcripts </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the fasta with transcript evidence. </font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--output-dir
			OUTPUT_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the outputs of the first annotation steps.
			Default step03_annotation_pipeline.V01/ </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/
			</font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasa-dir
			PASA_DIR   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"> Directory to keep all the pasa outputs. Default step03_annotation_pipeline.V01//protein_and_transcript_mappings/pasa/</font></font></p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Transdecoder
			parameters: </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--transdecoder-weights
			TRANSDECODER_WEIGHTS [TRANSDECODER_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to pasa transdecodergff3 output file when
			running EVM. Specify the weight for each EVM run separated
			by a space. Example 3 2 3</font></font></p>
		</td>
	</tr>
</table>

<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.11
Running EVM:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%"></p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"> <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome
			genome   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">     Path to the fasta genome.</font></font></p>
			<p align="justify" style="margin-bottom: 0in"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasa-dir
			PASA_DIR  </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">  Directory to keep all the pasa outputs. Default step03_annotation_pipeline.V01//protein_and_transcript_mappings/pasa/</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--spaln-weights
			SPALN_WEIGHTS [SPALN_WEIGHTS ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Weights
			given to spaln mappings when running EVM. Specify
			the weight for each EVM run separated by a space.
			Example 10 8 10 </font></font>
			</p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--update-dir
			UPDATE_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for annotation update step.
			Default step05_annotation_update.V01/</font></font></p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font size="3" style="font-size: 12pt">Evm
			parameters: </font>
			</p>
			<p align="justify">  <font size="3" style="font-size: 12pt">--evm-script
			EVM_SCRIPT </font>
			</p>
			<p align="justify">                        <font size="3" style="font-size: 12pt">Script
			to run EVM. Default /project/devel/aateam/bin/annotation_scripts/evm.sh
			</font>
			</p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt"><u>7.12
Running Annotation Update:</u></font></font></p>
<p style="margin-bottom: 0in; line-height: 100%; text-decoration: none">
<font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">When
running this step you need to make sure that the following parameters
are specified and that they have the correct values. The ones without
a default value (the inputs) are the ones that you are forced to
give, the rest will get the default value unless you change it. </font></font>
</p>
<p style="margin-bottom: 0in; line-height: 100%"></p>
<table width="100%" cellpadding="4" cellspacing="0">
	<col width="256*">
	<tr>
		<td width="100%" valign="top" style="border: 1px solid #000000; padding: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Inputs:</font></font></p>
			<p align="justify"> <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--genome
			genome  </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">   Path to the fasta genome.</font></font></p>
			<p align="justify" style="margin-bottom: 0in"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--EVM-inputs
			EVM_INPUTS </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for the EVM step. Default step04_EVM.V01/</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasa-dir
			PASA_DIR  </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">   Directory to keep all the pasa outputs. Default step03_annotation_pipeline.V01//protein_and_transcript_mappings/pasa/</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--pasadb
			pasadb   </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">     Name of the pasa database, it must coincide with the name
			in pasa_config. </font></font>
			</p>
			<p align="justify">   <font size="3" style="font-size: 12pt"><font face="Liberation Serif, serif">--update-config
			update_config </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the Pasa configuration file. </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--project-name
			project_name [project_name ...] </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Name
			of the project and version of the annotation space
			separated, to give the names to the final annotation
			output.</font></font></p>
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--geneid-parameters
			geneid_parameters </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Path
			to the geneid parameters file.</font></font></p>
		</td>
	</tr>
	<tr>
		<td width="100%" valign="top" style="border-top: none; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000; padding-top: 0in; padding-bottom: 0.04in; padding-left: 0.04in; padding-right: 0.04in">
			<p align="justify"><font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Outputs:</font></font></p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-version
			ANNOTATION_VERSION </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Version
			of the annotation process. Default 01</font></font></p>
			<p align="justify"> <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--annotation-step
			ANNOTATION_STEP </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Step
			of the annotation pipeline in the annotation process.
			Default 3 </font></font>
			</p>
			<p align="justify">  <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">--update-dir
			UPDATE_DIR </font></font>
			</p>
			<p align="justify">                        <font face="Liberation Serif, serif"><font size="3" style="font-size: 12pt">Directory
			to keep the files for annotation update step.
			Default step05_annotation_update.V01/</font></font></p>
		</td>
	</tr>
</table>
<p style="margin-bottom: 0in; line-height: 100%"><br>
</p>
<a href="#linkers"><CENTER><B>back</B></CENTER></a> <br>
</body>
</html>
